### PR TITLE
Adding animated progress dots

### PR DESCRIPTION
**Pull request by kwalker3690**
Implements #347 

@danoc @tomgenoni

### CODE REVIEW

**Comment by tomgenoni** [comment]
@kwalker3690 Could you update the changelog?

**Comment by danoc** [comment]
LGTM once the change log entry is there. It'd fall under `[Feature]`.

That said, I think we should ask the designers to communicate which should be used and when.

**Comment by danoc** [comment]
Also, lol at @chestone's commit showing up here somehow. (Hi Cheston!)

**Comment by kwalker3690** [comment]
Haha I know! Git struggles, but hey friend!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This file follows the format suggested by [Keep a CHANGELOG](https://github.com/olivierlacan/keep-a-changelog).
 
 ## [Unreleased][Unreleased]
+### Added
+- [Feature] Adding new style of loader, called `progress-dots. (#347)
+- [Patch] Renamed `progress.scss` to `progress-bar.scss`
```
Comment (danoc) [inline CHANGELOG.md]: This would be a `[Release]` since it will break any applications that may import `progress-bar`.

**Comment by danoc** [comment]
Also:

```
[18:33:00] 3 issues found in /home/travis/build/optimizely/oui/src/oui/partials/components/_progress-dots.scss
[18:33:00] partials/components/_progress-dots.scss:27 [W] LeadingZero: `.2` should be written with a leading zero as `0.2`
[18:33:00] partials/components/_progress-dots.scss:31 [W] LeadingZero: `.4` should be written with a leading zero as `0.4`
[18:33:00] partials/components/_progress-dots.scss:35 [W] SpaceBeforeBrace: Opening curly brace `{` should be preceded by one space
```

(There's a bug right now where Travis doesn't throw errors when lint breaks: #351)

**Pull request merged**